src/ShareRoutes.jsx

```diff
@@ -32,6 +33,79 @@ function Forward({appPrefix}) {
 }
 
 
+/**
+ * Check if input is a url
+ * @param {input} input
+ * @return {boolean} return true if url is found
+ */
+export function isURL(input) {
+  assertDefined(input)
+  return input.includes('https://') || input.includes('www')
+}
+
+
+/**
+ * Check if input is a valid url
+ * @param {Object} input
+ * @return {boolean} return true if url is found
+ */
+export function isValidModelURL(input) {
+  assertDefined(input)
+  try {
+    const url = new URL(input)
+    const parsed = new URL(url)
+    if (parsed.hostname !== 'github.com' &&
+        parsed.hostname !== 'www.github.com' &&
+        parsed.hostname !== 'raw.githubusercontent.com') {
+      return false
+    }
+    return true
+  } catch (e) {
+    return false
+  }
+}
+
+
+/**
+ * ParseIfcURL parses Github repository URLs or raw content URLs
+ * @param {string} url
+ * @return {string} Structured path to the model repositore
+ */
+export const parseIfcURL = (url) => {
```

Comment:
I think this should probably be limited to parseIfcRepoPath or smth, not the whole URL path.  This is because L88 below has hardcoded prefix for the app installation path, which should be avoided.  Caller should pull that off if possible.